src/commit.c

```diff
@@ -183,6 +183,35 @@ static int git_commit__create_internal(
 	return error;
 }
 
+/// Make git commit buffer similar to `git_commit__create_internal`
+/// but without write operation, `id` and `update_ref` arguments.
+int git_commit_create_buffer_for_parents_cb(
+	git_buf* out,
+	git_repository *repo,
+	const git_signature *author,
+	const git_signature *committer,
+	const char *message_encoding,
+	const char *message,
+	const git_oid *tree,
+	git_commit_parent_callback parent_cb,
+	void *parent_payload,
+	bool validate)
+ {
+	int error;
+	git_array_oid_t parents = GIT_ARRAY_INIT;
+
+	if ((error = validate_tree_and_parents(&parents, repo, tree, parent_cb, parent_payload, NULL, validate)) < 0)
```

Comment:
Hi!

Long time no C 😅 yet, this condition might be incorrectly interpret to bool, and then matched to 0

perhaps, PR may be a bit more welcome by libgit2 community, if this code can be simplified? 

e.g. 

```suggestion
        error = validate_tree_and_parents(&parents, repo, tree, parent_cb, parent_payload, NULL, validate);
	if (error < 0)
```

another take is, probably leaving out infamous `goto` and replace it with inframous `if {}`

```c
if (error >= 0) {
    GIT_BUF_WRAP_PRIVATE(...)
}
// cleanup...
```